src/scheduled_tasks.rs

```diff
@@ -23,33 +31,36 @@ pub fn setup(db_url: String, user_agent: String) -> Result<(), LemmyError> {
   // Setup the connections
   let mut scheduler = Scheduler::new();
 
-  let mut conn = PgConnection::establish(&db_url).expect("could not establish connection");
-
+  let mut conn_1 = PgConnection::establish(&db_url).expect("could not establish connection");
   let mut conn_2 = PgConnection::establish(&db_url).expect("could not establish connection");
+  let mut conn_3 = PgConnection::establish(&db_url).expect("could not establish connection");
+  let mut conn_4 = PgConnection::establish(&db_url).expect("could not establish connection");
 
-  active_counts(&mut conn);
-  update_banned_when_expired(&mut conn);
+  // Run on startup
+  active_counts(&mut conn_1);
+  update_hot_ranks(&mut conn_1);
+  update_banned_when_expired(&mut conn_1);
+  clear_old_activities(&mut conn_1);
+  update_instance_software(&mut conn_1, &user_agent);
 
-  // On startup, reindex the tables non-concurrently
-  // TODO remove this for now, since it slows down startup a lot on lemmy.ml
-  reindex_aggregates_tables(&mut conn, true);
+  // Update active counts every hour
   scheduler.every(CTimeUnits::hour(1)).run(move || {
-    let conn = &mut PgConnection::establish(&db_url)
-      .unwrap_or_else(|_| panic!("Error connecting to {db_url}"));
-    active_counts(conn);
-    update_banned_when_expired(conn);
-    reindex_aggregates_tables(conn, true);
-    drop_ccnew_indexes(conn);
+    active_counts(&mut conn_1);
+    update_banned_when_expired(&mut conn_1);
+  });
+
+  // Update hot ranks every 5 minutes
+  scheduler.every(CTimeUnits::minutes(5)).run(move || {
```

Comment:
I'm updating the hot ranks every 5 minutes. Since it only works on the last week of content, it shouldn't be too bad. 

Its a decaying time function which means that it *should be* live data, so I think the more frequently this is updated, the better.

Reply:
Seems like it would be worth making this configurable, so that it can be increased when server load is high.